### PR TITLE
:seedling:  upgrade to Kustomize configuration

### PR DESCRIPTION
**Pull request by khareyash05**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
To switch from Helm Charts to Kustomize

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by Iceber** [comment]
It is best to keep the contents of the deploy directory and the kustomize file in */kustomize* to ensure that */deploy* can still be deployed with the old documentation when kustomize is not fully available

**Comment by khareyash05** [comment]
> It is best to keep the contents of the _/deploy_ directory and the kustomize file in _/kustomize_ to ensure that _/deploy_ can still be deployed with the old documentation when kustomize is not fully available

will do that

File: kustomize/binding-apiserver/binding-apiserver.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
+apiVersion: v1
```
Comment (Iceber) [inline kustomize/binding-apiserver/binding-apiserver.yaml:1]: We shouldn't have to keep the yaml file for the build

File: kustomize/clusterpedia_apiserver.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
+apiVersion: v1
```
Comment (Iceber) [inline kustomize/clusterpedia_apiserver.yaml:1]: They can move into */kustomize/apiserver*

File: kustomize/kustomization.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ./binding-apiserver
+- ./apiserver
+- ./clustersynchro-manager
+- ./controller-manager
+- ./crds
+- ./internalstorage
```
Comment (Iceber) [inline kustomize/kustomization.yaml:10]: We can only choose one storage component when deploying, so we can't specify internalstorage here, but rather internalstorage/mysql or internalstorage/postgres

File: kustomize/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+
+resources:
+- ./binding-apiserver
+- ./apiserver
```
Comment (Iceber) [inline kustomize/kustomization.yaml:6]: apiserver and binding-apiserver cannot be deployed at the same time, we don't need to install `binding-apiserver` here.

File: kustomize/internalstorage/kustomization.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (Iceber) [inline kustomize/internalstorage/kustomization.yaml:1]: internalstorage doesn't need kustomize.yaml, we will only choose one of mysql or postgres in the installation.

**Review by Iceber — COMMENTED** [review]
The others look great. :thumbsup:



<br class="Apple-interchange-newline">

File: kustomize/setup.yaml [hunk 6]
```diff
@@ -0,0 +1,1110 @@
+apiVersion: v1
```
Comment (Iceber) [inline kustomize/setup.yaml:1]: We don't need this file generated by `kubectl kustomize`.

File: kustomize/apiserver/clusterpedia_apiserver.yaml [hunk 7]
```diff
@@ -0,0 +1,73 @@
+apiVersion: v1
```
Comment (Iceber) [inline kustomize/apiserver/clusterpedia_apiserver.yaml:1]: Last issue, this file is also generated by `kubectl kustomie`

Comment (khareyash05) [reply kustomize/apiserver/clusterpedia_apiserver.yaml]: Missed that . will update

**Review by Iceber — APPROVED** [review]
Thank you very much for your contribution

**Pull request merged**